src/axom/spin/BVH.hpp

```diff
@@ -275,12 +275,7 @@ class BVH
    * \param [out] counts stores the number of candidates for each ray
    * \param [out] candidates array of candidate IDs for each ray
    * \param [in] numRays the total number of rays
-   * \param [in] x0 array consisting the ray source point x-coordinates.
-   * \param [in] nx array consisting the ray normal x-components.
-   * \param [in] y0 array consisting the ray source point y-coordinates
-   * \param [in] ny array consisting the ray normal y-components
-   * \param [in] z0 array consisting the ray source point z-coorindates (in 3D)
-   * \param [in] nz array consisting the ray normal z-components (in 3D)
+   * \param [in] rays array of the rays to query against the BVH
```

Comment:
If a user already has the ray data as separate structs of arrays (e.g. `x0*`, `y0*`, `z0*`, `nx*`, `ny*`, `nz*`), what would be the best way for them to call this function?

Would they need to allocate a new array of `primal::Ray`s and construct them in a kernel?

Should we provide both options within the `BVH` class? ... and possibly use an internal `zip` function to compose the rays on the fly?

Note: Asking b/c I believe this is how some user codes already have their data laid out, and the extra allocation would be undesirable.

Reply:
That's definitely something worth considering.

I think it would be preferable to keep the API surface small. We might be able to make the `findRays` function templated on a `typename RayIterable`, so we could accept any object that returns a `primal::Ray` when it is dereferenced or accessed with `it[idx]` (and likewise for `findPoints/findBoundingBoxes`)

We could then define something akin to a `zip_iterator` within primal for `primal::Ray/Point/BoundingBox` so that user codes could pass in their SoA data without allocating another array for the primal primitives.

Reply:
That sounds like a really nice compromise! and a generally useful addition to primal/spin

Reply:
I've added the API change for accepting array-like types for the query objects in `find*()`, so they should be able to accept any object which is indexable with `operator[]`.

As for the `zip_iterator` types, I'll work on introducing that in a future PR. I've opened up an issue here: #604